### ISSUE TITLE
Project.json with compile/exclude does not build

### ISSUE THREAD

**Issue by nil4** [dotnet/cli#2841]
## Steps to reproduce
In an empty folder, run `dotnet new`. Change the generated `project.json` to:

```json
{
  "version": "1.0.0-*",
  "buildOptions": {
    "emitEntryPoint": true,
    "compile": {
        "exclude": ["node_modules"]
    },
  },
  "dependencies": {},
  "frameworks": {"net461": {}}
}
```
Run `dotnet restore`, then `dotnet build`.

## Expected  behavior
The project is built and an executable is generated in the `bin` folder, since `Program.cs` is not matched by the `exclude` pattern.

## Actual behavior
The project is not built and no message is displayed.

## Environment data
`dotnet --info` output:
```
.NET Command Line Tools (1.0.0-rc2-002611)

Product Information:
 Version:     1.0.0-rc2-002611
 Commit Sha:  bf8f0edd89

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.10586
 OS Platform: Windows
 RID:         win10-x64
```

**Comment by ajaybhargavb** [comment]
This happens due to the defaults being ignored when builtIns was not specified. This is fixed in https://github.com/dotnet/cli/issues/2819.

**Comment by piotrpMSFT** [comment]
Fixed by #2847
